airbyte-integrations/connectors/source-harvest/source_harvest/auth.py

```diff
@@ -4,14 +4,22 @@
 
 from typing import Any, Mapping
 
-from airbyte_cdk.sources.streams.http.auth import TokenAuthenticator
+from airbyte_cdk.sources.streams.http.auth import Oauth2Authenticator, TokenAuthenticator
 
 
-class HarvestTokenAuthenticator(TokenAuthenticator):
-    def __init__(self, token: str, account_id: str, account_id_header: str = "Harvest-Account-ID", **kwargs):
-        super().__init__(token, **kwargs)
+class HarvestMixin:
+    def __init__(self, *, account_id: str, account_id_header: str = "Harvest-Account-ID", **kwargs):
+        super().__init__(**kwargs)
         self.account_id = account_id
         self.account_id_header = account_id_header
 
     def get_auth_header(self) -> Mapping[str, Any]:
         return {**super().get_auth_header(), self.account_id_header: self.account_id}
+
+
+class HarvestTokenAuthenticator(HarvestMixin, TokenAuthenticator):
+    pass
+
+
+class HarvestOauth2Authenticator(HarvestMixin, Oauth2Authenticator):
+    pass
```

Comment:
Please add code comments about goals of this classes

Reply:
I have added docs for auth classes